src/BlazorIdentity.UI/ForgotPassword.razor

```diff
@@ -0,0 +1,81 @@
+﻿@typeparam TUser where TUser : class
+
+@using Microsoft.Extensions.Logging
+@using Microsoft.AspNetCore.WebUtilities
+@using System.Text
+@using System.Text.Encodings.Web
+
+@inject IBlazorSignInManager<TUser> SignInManager
+@inject IBlazorUserManager<TUser> UserManager
+@inject ILogger<ForgotPassword<TUser>> logger
+@inject NavigationManager NavManager
+@inject IBlazorEmailSender EmailSender
+
+<h4>Enter your email.</h4>
+
+<hr />
+
+<div class="row">
+	<EditForm class="col-5" EditContext="editContext" OnSubmit="@HandleSubmit">
+		<DataAnnotationsValidator />
+		<ValidationSummary />
+
+		<div class="mb-3">
+			<label for="Email" class="form-label">Email address</label>
+			<InputText id="Email" class="form-control" @bind-Value="forgotPasswordModel.Email" />
+		</div>
+
+		<div>
+			<button type="submit" class="btn btn-primary">Reset Password</button>
+		</div>
+
+	</EditForm>
+</div>
+
+@code {
+	private EditContext? editContext;
+	private ForgotPasswordModel forgotPasswordModel = new();
+	private ValidationMessageStore? messageStore;
+
+	[Parameter]
+	public string FailureMessage { get; set; } = "Login failed. Please check the email address and password.";
+
+	protected override void OnInitialized()
+	{
+		editContext = new(forgotPasswordModel);
+		messageStore = new(editContext);
+	}
+
+	private async Task HandleSubmit()
+	{
+		messageStore?.Clear();
+
+		if (editContext?.Validate() != true)
+		{
+			return;
+		}
+		var user = await UserManager.FindByEmailAsync(forgotPasswordModel.Email);
+
+		if (user == null || !(await UserManager.IsEmailConfirmedAsync(user)))
+		{
+			// Don't reveal that the user does not exist or is not confirmed
+			NavManager.NavigateTo("/Account/ForgotPasswordConfirmation");
```

Comment:
These components shouldn't navigate to pages that aren't in this library. For now, they should just raise events and let the hosting page handle that event and do the navigation.